src/platform/forms-system/src/js/helpers.js

```diff
@@ -701,3 +701,24 @@ export function transformForSubmit(
 
   return JSON.stringify(withoutViewFields, replacer) || '{}';
 }
+
+/**
+ * This function accepts a uiSchema object and returns its closest child object that contains the uiSchema entries
+ *
+ * Note: This is not intended for uiSchema child properties that have more than one sibling. Support for this is still a work in progress.
+
+ * @param {Object} uiSchema A nested `uiSchema`
+ * @returns {Object} The closest `uiSchema` object that holds the `uiSchema` keys
+ */
+export const getNestedUISchema = uiSchema => {
+  const uiSchemaKeys = Object.keys(uiSchema);
+  const isThisAUISchemaObj = !!uiSchemaKeys.find(uiSchemaKey =>
+    uiSchemaKey.includes('ui:'),
+  );
+  for (const uiSchemaKey of uiSchemaKeys) {
+    if (!isThisAUISchemaObj) {
+      return getNestedUISchema(uiSchema[uiSchemaKey]);
+    }
+  }
```

Comment:
There is a boolean const defined outside of the loop, and it is used inside of the loop to conditionally return something. The const does not change as we iterate through the loop. 

- `isThisAUISchemaObj` === `false`:  Inside the loop we take the first item and return the result of a function call. No other items are visited
- `isThisAUISchemaObj` === `true`:  Nothing at all happens inside the loop.

Can you explain why this is written with a loop?